sqlmesh/core/snapshot/evaluator.py

```diff
@@ -259,14 +259,25 @@ def create(
         """
         snapshots_with_table_names = defaultdict(set)
         tables_by_schema = defaultdict(set)
+        table_deployability: t.Dict[str, bool] = {}
         for snapshot in target_snapshots:
             if not snapshot.is_model or snapshot.is_symbolic:
                 continue
-            for is_deployable in (True, False):
+            deployability_flags = (
+                [True] if (not snapshot.is_managed or not snapshot.reuses_previous_version) else []
```

Comment:
I suggest we always have `True` here. Unconditionally

Reply:
This was only for managed tables to retain the logic. Which if I'm not mistaken since they don't have an existing deployable table it would lead to appear as missing, if you can confirm @erindru this as well?

Then in the case of having to create a deployable table it would lead in calling `create_managed_table` which we wouldn't want for a snapshot that reuses a previous version I think, But I can adapt the logic for managed tables in the `_create_snapshot` function so that is avoided and then always have `True` in the flags here

Reply:
Wdym? This is just for checking the table existence, right? 

>  which we wouldn't want for a snapshot that reuses a previous version I think

If it reuses the previous version than the deployable table **should** exist unless it was dropped externally, right?

Reply:
I see yes, I misunderstood how managed tables work because I was focusing on the test:
`test_create_managed_forward_only_with_previous_version_doesnt_clone_for_dev_preview`
because `get_data_objects` wasn't returning any existing deployable table. So I will patch this function in this test and then change to always check for missing deployable versions as you suggested